File: frontend/src/api/contests.js
/**
 * Contests API client
 */

const API_BASE = import.meta.env.VITE_API_URL || 'http://localhost:8080';

/**
 * Fetch upcoming radio contests
 * @param {number} days - Number of days to look ahead (default 30)
 * @returns {Promise<Object>} Contest data
 */
export const fetchUpcomingContests = async (days = 30) => {
  try {
    const response = await fetch(`${API_BASE}/api/v1/contests/upcoming?days=${days}`);
    if (!response.ok) {
      throw new Error(`API error: ${response.status}`);
    }
    return await response.json();
  } catch (err) {
    console.error('Error fetching contests:', err);
    throw err;
  }
};

/**
 * Fetch contests by band
 * @param {string} band - Band name (e.g., '20m', '40m')
 * @param {number} days - Number of days to look ahead (default 30)
 * @returns {Promise<Object>} Contests for specified band
 */
export const fetchContestsByBand = async (band, days = 30) => {
  try {
    const response = await fetch(`${API_BASE}/api/v1/contests/by-band/${band}?days=${days}`);
    if (!response.ok) {
      throw new Error(`API error: ${response.status}`);
    }
    return await response.json();
  } catch (err) {
    console.error(`Error fetching contests for band ${band}:`, err);
    throw err;
  }
};

/**
 * Fetch contests by mode
 * @param {string} mode - Mode name (e.g., 'CW', 'SSB', 'RTTY')
 * @param {number} days - Number of days to look ahead (default 30)
 * @returns {Promise<Object>} Contests for specified mode
 */
export const fetchContestsByMode = async (mode, days = 30) => {
  try {
    const response = await fetch(`${API_BASE}/api/v1/contests/by-mode/${mode}?days=${days}`);
    if (!response.ok) {
      throw new Error(`API error: ${response.status}`);
    }
    return await response.json();
  } catch (err) {
    console.error(`Error fetching contests for mode ${mode}:`, err);
    throw err;
  }
};

File: backend/app/routers/contests.py
"""Contests API endpoints"""
from fastapi import APIRouter, Query
from datetime import datetime, timezone
from app.services.contests import ContestsService

router = APIRouter(prefix="/api/v1/contests", tags=["contests"])

# Initialize service
contests_service = ContestsService()


@router.get("/upcoming")
async def get_upcoming_contests(days: int = Query(30, ge=1, le=365)):
    """
    Get upcoming radio contests
    
    Args:
        days: Number of days to look ahead (1-365, default 30)
        
    Returns:
        Dictionary with list of upcoming contests
    """
    print(f"📅 /contests/upcoming called with days={days}")
    result = await contests_service.get_upcoming_contests(days)
    return result


@router.get("/by-band/{band}")
async def get_contests_by_band(band: str, days: int = Query(30, ge=1, le=365)):
    """
    Get contests for a specific band
    
    Args:
        band: Band name (e.g., '20m', '40m', '2m')
        days: Number of days to look ahead (default 30)
        
    Returns:
        List of contests for the specified band
    """
    print(f"📅 /contests/by-band/{band} called")
    contests = await contests_service.get_contests_by_band(band, days)
    return {
        'band': band,
        'contests': contests,
        'count': len(contests),
        'updated': datetime.now(timezone.utc).isoformat()
    }


@router.get("/by-mode/{mode}")
async def get_contests_by_mode(mode: str, days: int = Query(30, ge=1, le=365)):
    """
    Get contests for a specific mode
    
    Args:
        mode: Mode name (e.g., 'CW', 'SSB', 'RTTY')
        days: Number of days to look ahead (default 30)
        
    Returns:
        List of contests for the specified mode
    """
    print(f"📅 /contests/by-mode/{mode} called")
    contests = await contests_service.get_contests_by_mode(mode, days)
    return {
        'mode': mode,
        'contests': contests,
        'count': len(contests),
        'updated': datetime.now(timezone.utc).isoformat()
    }
